ibis/backends/base/sql/alchemy/__init__.py

```diff
@@ -389,8 +389,13 @@ def _get_sqla_table(
     ) -> sa.Table:
         # If the underlying table (or more likely, view) has changed, remove it
         # to ensure a correct reflection
-        if autoload and self.inspector.has_table(name):
-            self.meta.remove(sa.Table(name, self.meta))
+        if (sa_table := self.meta.tables.get(name, None)) is not None and autoload:
+            # check if self._metadata and self.meta agree
+            # (ibis vs. sqlalchemy bookkeeping)
+            sa_cols = {col.name for col in sa_table.columns}
+            ibis_cols = {col for col, _ in self._metadata(name)}
```

Comment:
This call won't be particularly cheap since it's making a database query. I bet that it'll be awash with respect to `self.inspector.has_table` since that probably also makes at least one database query.

On the other hand, I'll be happy if this is in fact faster.